toml/src/main/kotlin/org/rust/toml/crates/local/CratesLocalIndexServiceImpl.kt

```diff
@@ -215,37 +220,57 @@ class CratesLocalIndexServiceImpl
         }
 
         val revTree = RevWalk(repository).parseCommit(ObjectId.fromString(currentHeadHash)).tree
+        val mapper = JsonMapper()
+            .registerKotlinModule()
+            .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES)
 
         TreeWalk(repository).use { treeWalk ->
             treeWalk.addTree(revTree)
             treeWalk.filter = filter
             treeWalk.isRecursive = true
             treeWalk.isPostOrderTraversal = false
 
+            val objectIds = mutableListOf<Pair<ObjectId, String>>()
             while (treeWalk.next()) {
                 if (treeWalk.isSubtree) continue
                 if (treeWalk.nameString == "config.json") continue
 
                 val objectId = treeWalk.getObjectId(0)
-                val loader = repository.open(objectId)
-                val versions = mutableListOf<CargoRegistryCrateVersion>()
-                val fileReader = BufferedReader(InputStreamReader(loader.openStream()))
-
-                fileReader.forEachLine { line ->
-                    if (line.isBlank()) return@forEachLine
-
+                objectIds.add(objectId to treeWalk.nameString)
+            }
+            val pool = Executors.newWorkStealingPool(2)
+            val future = supplyAsync(pool) {
+                objectIds
+                    .parallelStream()
+                    .map { (objectId, name) ->
+                        val loader = repository.open(objectId)
+                        val versions = mutableListOf<CargoRegistryCrateVersion>()
+                        val fileReader = BufferedReader(InputStreamReader(loader.openStream()))
+
+                        fileReader.forEachLine { line ->
+                            if (line.isBlank()) return@forEachLine
+
+                            try {
+                                versions.add(crateFromJson(line, mapper))
+                            } catch (e: Exception) {
+                                LOG.warn("Failed to parse JSON from ${treeWalk.pathString}, line ${line}: ${e.message}")
+                            }
+                        }
+                        name to CargoRegistryCrate(versions)
+                    }
+                    .collect(Collectors.toList())
+            }
+            try {
+                val crateList = future.join()
```

Comment:
By the way, the `try` block could be closed right after `future.join()` invocation since we doesn't use the pool later